core/src/transaction_meta.rs

```diff
@@ -44,7 +44,7 @@ impl TransactionMeta {
     }
 
     pub fn is_spent(&self, index: usize) -> bool {
-        self.output_spent.get(index).unwrap_or(false)
+        self.output_spent.get(index).unwrap_or(true)
```

Comment:
Why return `true` when index is out of bounds?

Reply:
This output will not appear, and can not be spent, just like being spent.